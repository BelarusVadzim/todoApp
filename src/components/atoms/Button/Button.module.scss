@import "src/variables.scss";

.button {
  color: $secondary-font-color;
  cursor: pointer;
  font-family: $main-font;
  display: inline-block;
  font-size: $font-size !important;
  
  &:hover {
    color: rgb(80 80 80);
  }
}
