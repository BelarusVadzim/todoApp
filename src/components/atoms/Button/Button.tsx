import React, { useMemo } from 'react';
import style from './Button.module.scss';
import { PropsWithClassName } from 'types';

type ButtonProps = PropsWithClassName & {
  value: string;
  onClick?: () => void;
  glyph?: boolean;
};

const Button: React.FC<ButtonProps> = ({
  value,
  onClick,
  glyph,
  className,
}) => {
  let combinedClasses = useMemo(
    () => `${glyph ? 'material-icons' : ''} ${style.button}`,
    [glyph],
  );

  //ToDo: put in separated function
  combinedClasses = className
    ? `${combinedClasses} ${className}`
    : combinedClasses;

  return (
    <div onClick={onClick} className={combinedClasses}>
      {value}
    </div>
  );
};

export default Button;
