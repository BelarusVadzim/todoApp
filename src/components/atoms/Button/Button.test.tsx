import { render } from '@testing-library/react';
import Button from './Button';

describe('<Button />', () => {
  it('should render properly', () => {
    const component = render(<Button value="buttonValue" />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when glyph className specified should render properly', () => {
    const component = render(
      <Button glyph={true} value="buttonValue" className="testClassName" />
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
