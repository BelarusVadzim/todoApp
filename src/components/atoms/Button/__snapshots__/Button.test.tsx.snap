// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
<DocumentFragment>
  <div
    class=" button"
  >
    buttonValue
  </div>
</DocumentFragment>
`;

exports[`<Button /> when glyph className specified should render properly 1`] = `
<DocumentFragment>
  <div
    class="material-icons button testClassName"
  >
    buttonValue
  </div>
</DocumentFragment>
`;
